### ISSUE TITLE
Property specifiers expected type of array but got null

### ISSUE THREAD

**Issue by CHNB128** [th3rdwave/react-native-safe-area-context#198]
I launch application one day and got this error while bundling: 
```
[Fri Jun 11 2021 11:45:38.105]  BUNDLE  ./index.android.js 

error: node_modules/react-native-safe-area-context/src/SafeArea.types.ts: /home/gnome/src/urbest/z-frontend/node_modules/react-native-safe-area-context/src/SafeArea.types.ts: Property specifiers expected type of array but got null
```

I can't determinate what went wrong. May be I change something in usage of `SafeAreaView` so it got null where should be array or something else.

**Comment by jbolter** [comment]
I just started seeing this as well and I'm wondering what changed 🤔

**Comment by khattab93** [comment]
Me too. Could anyone help me fix this?

**Comment by taylorkline** [comment]
I see this after upgrading from `react-native` v0.61.5 to v0.62.2.

**Comment by taylorkline** [comment]
@CHNB128 @jbolter @khattab93 Can each of you share information about your `react-native` and `react-navigation` versions?

**Comment by khattab93** [comment]
@taylorkline "react-native": "0.63.4" and I am using react native navigation

**Comment by CHNB128** [comment]
@taylorkline 
``` 
"react-native-safe-area-context": "^3.2.0",
"react-native": "0.63.4",
```
And we don't use `react-navigation`, because we have to outdated project, so it's a lot to change to migrate to it.

**Comment by CHNB128** [comment]
I have red some resources about similar errors: 
https://github.com/nrwl/nx/issues/3202#issuecomment-646824968
https://github.com/babel/babel/issues/4342#issue-175602831
https://stackoverflow.com/a/35404879
https://github.com/facebook/regenerator/issues/391
https://github.com/kentcdodds/babel-plugin-codegen/issues/17
And it's look like error during babel compilation, even if this error happens during bundling. 
So, can @taylorkline @khattab93 @harking also provide some information about your babel setup, which version of babel related dependencies you using (whole `package.json` dependencies section would be grate)?

**Comment by taylorkline** [comment]
Note I upgraded to React Native `v0.64.2` in hopes that would fix the issue - it didn't.

Current dependencies:

```
"dependencies": {
    "@bugsnag/react-native": "^7.5.2",
    "@invertase/react-native-apple-authentication": "^2.1.1",
    "@mapbox/polyline": "^1.0.0",
    "@react-native-community/async-storage": "^1.6.1",
    "@react-native-community/cameraroll": "^4.0.0",
    "@react-native-community/geolocation": "^2.0.2",
    "@react-native-community/google-signin": "^4.0.3",
    "@react-native-community/masked-view": "^0.1.10",
    "@react-native-community/netinfo": "^3.2.1",
    "@react-native-community/push-notification-ios": "^1.8.0",
    "@react-native-firebase/analytics": "11.4.1",
    "@react-native-firebase/app": "11.4.1",
    "axios": "^0.18.0",
    "axios-mock-adapter": "^1.15.0",
    "date-fns": "^1.29.0",
    "lodash": "^4.17.10",
    "prop-types": "^15.7.2",
    "qs": "^6.5.2",
    "react": "17.0.1",
    "react-native": "0.64.2",
    "react-native-actionsheet": "^2.4.2",
    "react-native-animatable": "^1.3.3",
    "react-native-device-info": "^2.3.2",
    "react-native-fbsdk-next": "^4.0.0",
    "react-native-gesture-handler": "^1.10.3",
    "react-native-idfa": "^4.1.0",
    "react-native-keyboard-aware-scroll-view": "^0.9.2",
    "react-native-keyboard-aware-scrollview": "^2.0.0",
    "react-native-linear-gradient": "^2.4.0",
    "react-native-localization": "2.1.6",
    "react-native-map-link": "^2.4.5",
    "react-native-maps": "^0.27.1",
    "react-native-permissions": "2.2.2",
    "react-native-platform-touchable": "^1.1.1",
    "react-native-push-notification": "7.3.1",
    "react-native-reanimated": "1.13.1",
    "react-native-safe-area-context": "^3.2.0",
    "react-native-screens": "^2.18.1",
    "react-native-sensitive-info": "^5.5.0",
    "react-native-snap-carousel": "3.7.2",
    "react-native-splash-screen": "3.2.0",
    "react-native-store-review": "^0.1.5",
    "react-native-video": "^4.4.4",
    "react-native-view-shot": "3.1.2",
    "react-native-webview": "^10.8.3",
    "react-navigation": "^4.0.0",
    "react-navigation-drawer": "^2.7.0",
    "react-navigation-stack": "^2.10.4",
    "react-redux": "^5.1.2",
    "redux": "^4.0.0",
    "redux-thunk": "^2.3.0",
    "uuid": "^3.3.2"
},
"devDependencies": {
    "@babel/core": "^7.12.9",
    "@babel/runtime": "^7.12.5",
    "babel-eslint": "^8.2.3",
    "babel-jest": "^26.6.3",
    "deep-freeze": "^0.0.1",
    "eslint": "^4.19.1",
    "eslint-plugin-import": "^2.22.1",
    "eslint-plugin-react": "^7.7.0",
    "eslint-plugin-react-hooks": "^3.0.0",
    "eslint-plugin-react-native": "^3.2.1",
    "jest": "^26.6.3",
    "jest-junit": "^6.3.0",
    "metro-react-native-babel-preset": "^0.64.0",
    "patch-package": "^6.2.2",
    "prettier": "1.12.1",
    "react-native-clean-project": "^3.4.0",
    "react-test-renderer": "17.0.1",
    "redux-logger": "^3.0.6",
    "redux-mock-store": "^1.5.3"
},
```

**Comment by CHNB128** [comment]
> Note I upgraded to React Native v0.64.2 in hopes that would fix the issue - it didn't.

I also thought about it, but same. 
Thanks for your replay. It's great to not be alone in this :)

**Comment by CHNB128** [comment]
```
  "dependencies": {
    "@cljs-oss/module-deps": "^1.1.1",
    "@react-native-community/async-storage": "^1.6.2",
    "@react-native-community/netinfo": "^4.2.1",
    "@react-native-community/push-notification-ios": "^1.8.0",
    "@react-native-firebase/analytics": "^10.8.1",
    "@react-native-firebase/app": "^10.8.1",
    "@react-native-firebase/crashlytics": "^10.8.1",
    "@react-native-firebase/messaging": "^10.8.1",
    "@sentry/react-native": "^1.0.5",
    "amplitude-js": "^5.8.0",
    "create-react-class": "15.6.3",
    "cypress-file-upload": "^4.0.7",
    "detox": "^16.7.1",
    "hermes-engine": "^0.2.1",
    "immutable": "^4.0.0-rc.12",
    "mocha": "^7.2.0",
    "react": "16.13.1",
    "react-dom": "16.9.0",
    "react-native": "0.63.4",
    "react-native-android-location-enabler": "^1.2.2",
    "react-native-calendars": "^1.204.0",
    "react-native-charts-wrapper": "^0.5.7",
    "react-native-code-push": "^7.0.0",
    "react-native-config": "^1.4.2",
    "react-native-device-info": "^3.1.1",
    "react-native-dialogs": "^1.0.5",
    "react-native-document-picker": "^3.2.4",
    "react-native-fast-image": "^8.3.4",
    "react-native-geolocation-service": "^5.2.0",
    "react-native-gesture-handler": "^1.10.3",
    "react-native-image-crop-picker": "^0.36.2",
    "react-native-image-zoom-viewer": "^3.0.1",
    "react-native-interactable": "https://github.com/wix/react-native-interactable.git#master",
    "react-native-keyboard-aware-scroll-view": "^0.9.4",
    "react-native-localize": "^1.2.0",
    "react-native-maps": "^0.28.0",
    "react-native-modal": "11.10.0",
    "react-native-orientation": "^3.1.3",
    "react-native-permissions": "^3.0.4",
    "react-native-push-notification": "^7.3.1",
    "react-native-reanimated": "^1.2.0",
    "react-native-root-toast": "^3.1.2",
    "react-native-safe-area-context": "^3.2.0",
    "react-native-screens": "2.0.0",
    "react-native-snap-carousel": "^3.8.0",
    "react-native-splash-screen": "^3.2.0",
    "react-native-svg": "12.1.0",
    "react-navigation": "^4.0.5",
    "react-navigation-stack": "^1.7.3",
    "react-navigation-tabs": "^2.5.2",
    "supercluster": "^6.0.1",
    "xmlhttprequest": "^1.8.0"
  },
  "devDependencies": {
    "@babel/core": "^7.8.4",
    "@babel/runtime": "^7.8.4",
    "@react-native-community/cli": "4.14.0",
    "@sentry/cli": "^1.49.0",
    "autoprefixer": "^9.6.4",
    "babel-jest": "^25.1.0",
    "bundlesize": "^0.17.2",
    "cssnano": "^4.1.10",
    "cypress": "4.6.0",
    "cypress-plugin-retries": "^1.5.2",
    "jest": "^25.1.0",
    "jetifier": "^1.6.2",
    "metro-react-native-babel-preset": "^0.59.0",
    "mqpacker": "^7.0.0",
    "patch-package": "^6.0.5",
    "postcss-cli": "^6.1.3",
    "react-test-renderer": "16.13.1",
    "serve": "^11.3.2"
  },
```

**Comment by CHNB128** [comment]
I created [discussion](https://github.com/babel/babel/discussions/13474) about this in https://github.com/babel/babel, may be some one will help us.

**Comment by nicolo-ribaudo** [comment]
Babel maintainer here. Does anyone have a complete stack trace or a repository I can clone and run a build command to reproduce the error?

**Comment by CHNB128** [comment]
> Babel maintainer here. Does anyone have a complete stack trace or a repository I can clone and run a build command to reproduce the error?

I can't get full stack trace. It would be great actually, if I can get it, but metro collapse it and we can get only this error line, even when I provide `--verbose` option to it. 
About to provide full repo to you, I also can't do it. It under commercial protection.

**Comment by nicolo-ribaudo** [comment]
The error is thrown at line 158 of https://unpkg.com/browse/@babel/types@7.14.5/lib/definitions/utils.js (it might be a slightly different line if you have a different `@babel/types` version).

Could you try editing it like this? Maybe `metro` will give us a stack trace.
```diff

function assertValueType(type) {
  function validate(node, key, val) {
    const valid = getType(val) === type;

    if (!valid) {
+      Error.stackTraceLimit = 100;
+      console.trace("ERROR HERE!");
      throw new TypeError(`Property ${key} expected type of ${type} but got ${getType(val)}`);
    }
  }

  validate.type = type;
  return validate;
}
```

**Comment by CHNB128** [comment]
Bit more about it. 
it's `node` value of `validate` function, if it will help. 
```
transform[stdout]: {
transform[stdout]:   type: 'ExportNamedDeclaration',
transform[stdout]:   declaration: null,
transform[stdout]:   specifiers: null,
transform[stdout]:   source: null
transform[stdout]: }
```

**Comment by nicolo-ribaudo** [comment]
Thanks, I'm trying to reproduce the bug using that stack trace :+1:

**Comment by nicolo-ribaudo** [comment]
Can you check the version of `/home/gnome/src/urbest/z-frontend/node_modules/@babel/types`?

**Comment by CHNB128** [comment]
The highest version I have in `.lock` file is:
```
  dependencies:
    "@babel/types" "^7.14.5"
```

**Comment by nicolo-ribaudo** [comment]
Can you check what `/home/gnome/src/urbest/z-frontend/node_modules/@babel/types/package.json` says? Looking at the code, this error should have been fixed in `@babel/types` 7.8.0 :thinking:

**Comment by CHNB128** [comment]
Yes, it's `7.14.5` in it.

**Comment by nicolo-ribaudo** [comment]
Can you also try logging `args` (other than `node`) for that `validate` call?
It should be an empty array, but I _think_ that for some reason it's something like `[null, null, null]`.

**Comment by khattab93** [comment]
I solved it by removing `yarn.lock` and `package-lock.json` and then running ```yarn install```

**Comment by CHNB128** [comment]
So, as I understood problem was that `yarn` or `npm` incorrectly resolved dependencies, which produce this strange error. 
Thanks all of you for helping with it. 
Think issue can be closed soon.

**Comment by taylorkline** [comment]
We have a workaround, not a solution. I disagree with closing this issue. Something is wrong with some dependencies somewhere.

@nicolo-ribaudo My babel types version is actually `7.5.5`.

And here are the arguments to the validate call:

```
transform[stdout]: node {
transform[stdout]:   type: 'ExportNamedDeclaration',
transform[stdout]:   declaration: null,
transform[stdout]:   specifiers: null,
transform[stdout]:   source: null
transform[stdout]: }
transform[stdout]: key specifiers
transform[stdout]: val null
transform[stdout]: type array
```

**Comment by taylorkline** [comment]
As a less heavy-handed workaround, I specifically removed the `@babel/types` entries from my `yarn.lock` and re-ran yarn. Now my `@babel/types` resolved to `7.14.5` and all works correctly.

**Comment by uzegonemad** [comment]
I had two entries for `@babel/types` in my `yarn.lock` file which resolved to 7.6.1 and 7.15.0. Instead of manually modifying my lock file, I ran `npx yarn-deduplicate` to handle the multiple entries. Afterward the only entry was for 7.15.0 which resolved this error.

**Comment by jacobp100** [comment]
This is for v3 of the library, and we're on v4. Please re-open a new issue if you're still facing issues
